### PR TITLE
Containerized local development

### LINKED ISSUE #6: feature: Docker setup for local development
The readme says that the playground only runs on linux so I was thinking about creating a Docker and a VSCode DevContainer setup to make it possible to run this on any system using containers

If this sounds good I can start working on it asap

### PR DESCRIPTION
**Pull request by ulises-jeremias**
This PR closes #6 

* * *

# Vlang Playground

This repository contains the source code for V's playground.

### Using Docker Compose

```bash
git clone https://github.com/vlang/playground
cd playground
docker-compose up -d
```

then access the playground at <http://localhost:5555>

### Using VSCode DevContainers

1. Install Docker
2. Install [Visual Studio Code](https://code.visualstudio.com/)
3. Install the [Remote Development](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack) extension for VS Code
4. Clone <https://github.com/vlang/playground>
5. Create your application within a container (see gif below)

Done. And your system will remain "clean".

vscode-open-in-container

Then just run

```sh
v run server.v
```

then access the playground at <http://localhost:5555>

## Run the playground locally (not recommended)

> NOTE: This is not recommended, as it requires you to install V and all of its dependencies manually. Only works on Linux.

### Install Dependencies

> We use isolate to sandbox the playground, so you need to install it first.

```bash
git clone https://github.com/ioi/isolate /tmp/isolate
cd /tmp/isolate
make isolate isolate-check-environment
make install
```

### Run the server

```bash
git clone https://github.com/vlang/playground
cd playground
v run server.v
```

then access the playground at <http://localhost:5555>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,60 @@
-# playground
+# Vlang Playground
 
-Source code for V's playground.
+This repository container the source code for V's playground.
```
Comment (spytheman) [inline README.md:3]: This repository contains the source code for V's playground.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+services:
+  playground:
+    command: bash -c "cd /usr/src/app && v run server.v"
+    build:
+      context: .
+      dockerfile: Dockerfile
+    ports:
+      - "5555:5555"
+    volumes:
+      - .:/usr/src/app
```
Comment (spytheman) [inline docker-compose.yml:12]: I am just curious, is there a convention or requirement for using `/usr/src/app`?

File: .devcontainer/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      # - /var/run/docker.sock:/var/run/docker.sock 
+
+    # Uncomment the next four lines if you will use a ptrace-based debugger like C++, Go, and Rust.
+    # cap_add:
+    #   - SYS_PTRACE
+    # security_opt:
+    #   - seccomp:unconfined
+
+    # Overrides default command so things don't shut down after the process ends.
+    command: /bin/sh -c "while sleep 1000; do :; done"
```
Comment (spytheman) [inline .devcontainer/docker-compose.yml:37]: Why 1000? That would be ~17 minutes. Would it stop after that time?

Comment (spytheman) [reply .devcontainer/docker-compose.yml:37]: oh, now I get it - there is a loop ... so it will sleep forever at 17 minutes slices.

**Comment by spytheman** [comment]
The .gif is over 1MB; if it is converted to .mp4 it may be much smaller.

File: docker-compose.yml (hunk 2 above)
Comment (ulises-jeremias) [reply docker-compose.yml:12]: Yeah, there is a convention when creating containers for apps to use that dir, not sure why 😅

File: README.md (hunk 1 above)
Comment (ulises-jeremias) [reply README.md]: Haha sorry, was the autocomplete, done!

File: .devcontainer/docker-compose.yml (hunk 3 above)
Comment (ulises-jeremias) [reply .devcontainer/docker-compose.yml:37]: Exactly that!

**Comment by ulises-jeremias** [comment]
@spytheman I'll try to push a MP4 file as then I'll ask for review again 👌🏻

**Comment by ulises-jeremias** [comment]
@spytheman solved using an autogenerated link provided by github

File: docker-compose.yml (hunk 2 above)
Comment (spytheman) [reply docker-compose.yml:12]: thanks

**Review by spytheman — APPROVED** [review]

**Comment by ulises-jeremias** [comment]
@medvednikov will be great if you can add the `hacktoberfest-accepted` label before merging!

**Pull request merged**